adrs/README.md

```diff
@@ -0,0 +1,21 @@
+# Architecture Decision Records (ADRs)
+An architecture decision record (ADR) is a document that captures an important architecture decision made along with its context and consequences. We strive to keep these records small and lean.
+
+### Goal
+We want to record important design decisions to help current and future contributors of the library to understand the reasons for doing things a certain way.
+
+
+### Lightweight Process
+
+- Create a new branch on this repository and copy `adrs/0000-template.md` to a new file with the next free number.
+- Modify the template to add details about your proposal and add any supporting picture under the `adrs/images` folder.
```

Comment:
Given this repo's history with images, I don't know if I want to encourage committing images to the repo as it's another thing we have to keep track of size-wise. Maybe we can punt this down the line and encourage images to be added the the GH PR? When the ADR is merged, the PR number should be in the commit message, so there should still be a canonical link to it.

Reply:
I understand. Maybe we could incentivise the use of ASCII diagrams for the time being and decide how we want to host images when there is a first use case.

Reply:
Ooh excellent idea, especially since GH has native support for [mermaid](https://mermaid-js.github.io/mermaid/#/)